test/core/security/credentials_test.cc

```diff
@@ -2451,21 +2500,61 @@ TEST(CredentialsTest,
   HttpRequest::SetOverride(nullptr, nullptr, nullptr);
 }
 
+TEST(
+    CredentialsTest,
+    TestExternalAccountCredsSuccessWithServiceAccountImpersonationAndCustomTokenLifetime) {
```

Comment:
Can you add a test that shows the error is thrown when the lifetime value provided is over 43200s? And that we use the default value of 3600s when no lifetime is provided.